shared/src/webMain/kotlin/org/jetbrains/kotlinconf/NotificationManager.web.kt

```diff
@@ -1,17 +1,62 @@
 package org.jetbrains.kotlinconf
 
-// TODO implement
+private external object Notification {
+    fun requestPermission(callback: (String) -> Unit)
+}
+
+private fun registerNotificationByServiceWorker(delay: Long, title: String, message: String): Unit =
+    js(
+        """{
+        if (typeof navigator === "undefined" || navigator.serviceWorker == null || !navigator.serviceWorker.ready) return;
+        navigator.serviceWorker.ready.then(function (registration) {
+          registration.active.postMessage({
+            command: 'register-notification',
+            title: title,
+            body: message,
+            delay: Number(delay),
+          });
+        });
+        }"""
+    )
+
+private fun cancelNotificationByServiceWorker(title: String): Unit =
+    js(
+        """{
+        if (typeof navigator === "undefined" || navigator.serviceWorker == null || !navigator.serviceWorker.ready) return;
+        navigator.serviceWorker.ready.then(function (registration) {
+          registration.active.postMessage({
+            command: 'cancel-notification',
+            title: title,
+          });
+        });
+        }"""
+    )
+
 actual class NotificationManager actual constructor(
     context: ApplicationContext
 ) {
+    private var notificationAllowed = false
+
     actual fun requestPermission() {
+        Notification.requestPermission {
+            if (it == "granted") {
+                notificationAllowed = true
```

Comment:
Notification will be shown only on the first open, after reopen/refresh `notificationAllowed` will be `false`.

Reply:
As far as I remember, it should be saved by browser for a specific host

Reply:
But it's runtime value, and it will not be persisted between runs